native/packages/react-native-bpk-component-image/src/BpkImage.js

```diff
@@ -0,0 +1,170 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  type StyleObj,
+  StyleSheet,
+  View,
+  ViewPropTypes,
+  Animated,
+} from 'react-native';
+import BpkSpinner from 'react-native-bpk-component-spinner';
+
+import {
+  borderRadiusSm,
+  colorGray300,
+  animationDurationBase,
+} from 'bpk-tokens/tokens/base.react.native';
+
+const styles = StyleSheet.create({
+  outer: {
+    width: '100%',
+    height: '100%',
+  },
+  withBorderRadius: {
+    borderRadius: borderRadiusSm,
+    overflow: 'hidden',
+  },
+  image: { width: '100%', height: '100%', position: 'absolute' },
+  spinnerView: {
+    width: '100%',
+    height: '100%',
+    backgroundColor: colorGray300,
+    justifyContent: 'center',
+    alignItems: 'center',
+    position: 'absolute',
+  },
+});
+
+export type Props = {
+  style: StyleObj,
+  inView: ?boolean,
+  borderRadius: ?boolean,
+  loaded: ?boolean,
+  alt: string,
+  source: string,
+  onLoad: ?func,
+};
```

Comment:
A few small Flow issues:

* Convention with is required first (alphabetised), then optional (alphabetised). Same as we do with prop types.
* We avoid using `func` because it's too vague. Flow lets you define exactly what arguments a function takes, plus the return type. As `onLoad` takes no arguments and can return anything (it's up to the consumer), the type would be `?() => mixed`.
* Flow types don't map 1:1 with prop types. Rule of thumb is that it's only optional if the `defaultProps` declaration is `null`. In this case, the `boolean` types all have non-null defaults so they're not optional. As `style` **does** default to `null`, it should be optional.

Put it all together and you get:

```
type Props = {
  alt: string,
  borderRadius: boolean,
  inView: boolean,
  loaded: boolean,
  source: string,
  onLoad: ?() => mixed,
  style: ?StyleObj,
};
```